### PR TITLE
Improved coverage of dcuregression scenarios

### PR DESCRIPTION
**Pull request by CPage-Ten10**

### CODE REVIEW

File: src/test/resources/features/dcu/DCUDataInput.feature [hunk 1]
```diff
@@ -50,36 +56,56 @@ Feature: DCU Data Input
       | Correspondence Received Date | Correspondence Received |
 
   @DCURegression
```
Comment (CaseyProsser) [inline src/test/resources/features/dcu/DCUDataInput.feature:58]: I dont think the two below scenarios didnt need to be changed, as they arent workflow tests, and I think we would be covered just testing the correspondent functionality on one workflow. The workflow test for this stage will need a correspondent to be added anyway.

File: src/test/resources/features/dcu/DCUDataInput.feature [hunk 2]
```diff
@@ -90,55 +116,79 @@ Feature: DCU Data Input
...
   @OtherTests
   Scenario: User edits an existing correspondent
-    When I fill all mandatory fields on the "Data Input" page with valid data
+    When I create a "MIN" case and move it to the "Data Input" stage
+    And I load and claim the current case
+    And I fill all mandatory fields on the "Data Input" page with valid data
     And I click the "Continue" button
     And I add "Nicola Sturgeon" MP as a correspondent
     And I edit the primary correspondents name
     Then the correspondents name should be updated
```
Comment (CaseyProsser) [inline src/test/resources/features/dcu/DCUDataInput.feature:171]: Does the below scenario not overlap with the deadlines feature? If it doesnt than I think we should consider refactoring this scenario into Deadline.

File: src/test/resources/features/dcu/DCUDataInput.feature [hunk 3]
```diff
@@ -90,55 +116,79 @@ Feature: DCU Data Input
...
-  Scenario: User creates a MIN case and checks that the stage deadlines are correct
-    Then the stage deadline dates for a "MIN" case are correct
+  Scenario Outline: User creates a DCU case and checks that the stage deadlines are correct
+    When I create a "<caseType>" case and move it to the "Data Input" stage
+    And I load and claim the current case
+    Then the stage deadline dates for a "<caseType>" case are correct
+    Examples:
+      | caseType  |
+      | MIN       |
+      | TRO       |
```
Comment (CaseyProsser) [inline src/test/resources/features/dcu/DCUDataInput.feature:181]: Again, I dont think this requires parameterisation as this isnt strictly a workflow test. Does this feature not have a @DCUWorkflow tagged scenario that tests sending a case from Data Input to Markup stage? Thats the one I thought we would alter

File: src/test/resources/features/dcu/Dispatch.feature [hunk 4]
```diff
@@ -47,5 +52,5 @@ Feature: Dispatch
     Then the case should be moved to the "Private Office Approval" stage
     Examples:
       | caseType |
-      | MIN  |
-      | DTEN |
```
Comment (CaseyProsser) [inline src/test/resources/features/dcu/Dispatch.feature:51]: Do we need an additional scenario for rejecting a TRO case at dispatch? Similar to this one but with an assertion appropriate to the TRO workflow

File: src/test/resources/features/dcu/InitialDraft.feature [hunk 5]
```diff
@@ -13,15 +13,19 @@ Feature: Initial Draft
     Then an error message appears instructing me to add rejection reasons
```
Comment (CaseyProsser) [inline src/test/resources/features/dcu/InitialDraft.feature:14]: For below scenario, does DTEN not have telephone option?

File: src/test/resources/features/dcu/InitialDraft.feature [hunk 6]
```diff
@@ -33,8 +37,8 @@ Feature: Initial Draft
     Then I see an error message instructing me to enter call notes
```
Comment (CaseyProsser) [inline src/test/resources/features/dcu/InitialDraft.feature:38]: Same here, can we not include DTEN?

File: src/test/resources/features/dcu/NoResponseNeededConfirmation.feature [hunk 7]
```diff
@@ -3,23 +3,41 @@ Feature: No Response Needed Confirmation
...
+    When I create a "<caseType>" case and move it to the "No Response Needed Confirmation" stage
+    And I load and claim the current case
+    And I click the NRN no radio button
     And I click the "Finish" button
     Then the case should be moved to the "Markup" stage
+    Examples:
+      | caseType  |
+      | MIN       |
+      | TRO       |
+      | DTEN      |
```
Comment (CaseyProsser) [inline src/test/resources/features/dcu/NoResponseNeededConfirmation.feature:32]: I dont think we need to parameterise the validation scenarios, as those share the same page configuration for all workflows.

File: src/test/resources/features/dcu/TransferConfirmation.feature [hunk 8]
```diff
@@ -3,22 +3,41 @@ Feature: TransferConfirmation
...
+    When I create a "<caseType>" case and move it to the "Transfer Confirmation" stage
+    And I load and claim the current case
+    And I click the confirm transfer no radio button
     And I click the "Finish" button
     Then the case should be moved to the "Markup" stage
+    Examples:
+      | caseType  |
+      | MIN       |
+      | TRO       |
+      | DTEN      |
```
Comment (CaseyProsser) [inline src/test/resources/features/dcu/TransferConfirmation.feature:32]: Same with the below, we dont need to turn this into a scenario outline

**Review by CaseyProsser — CHANGES_REQUESTED** [review]
Most of this is the change we need, but I think you might have amended some scenarios we don't need to run for each DCU workflow type.

File: src/test/resources/features/dcu/DCUDataInput.feature (hunk 2 above)
Comment (CPage-Ten10) [reply src/test/resources/features/dcu/DCUDataInput.feature:171]: Found duplicate scenario in Deadlines feature, removed the DCUDataInput instance of scenario

File: src/test/resources/features/dcu/Dispatch.feature (hunk 4 above)
Comment (CPage-Ten10) [reply src/test/resources/features/dcu/Dispatch.feature:51]: Added separate scenario for TRO workflow being rejected to initial draft from dispatch

File: src/test/resources/features/dcu/InitialDraft.feature (hunk 5 above)
Comment (CPage-Ten10) [reply src/test/resources/features/dcu/InitialDraft.feature:14]: DTEN don't have an option to select a response channel at Initial draft

**Review by CaseyProsser — APPROVED** [review]

**Pull request merged**